File: templates/new-module/__manifest__.py
"""Module manifest - defines metadata for the module loader."""

from typing import Literal, TypedDict


class ModuleManifest(TypedDict):
    """Type definition for module manifest structure."""

    name: str
    version: str
    main_route: str
    type: Literal["App", "System"]
    depends: list[str]
    icon_class: str
    color: str
    description: str
    long_description: str


# Rename 'modulename' to your actual module name
manifest: ModuleManifest = {
    # Required fields
    "name": "ModuleName",  # Display name (PascalCase)
    "version": "1.0",  # Semantic version
    "main_route": "/modulename",  # URL prefix (lowercase)
    "type": "App",  # "App" (visible) or "System" (hidden)
    "depends": ["core"],  # Module dependencies
    # Display fields
    "icon_class": "fa-solid fa-cube",  # FontAwesome icon class
    "color": "#007bff",  # Theme color (hex)
    "description": "Short description of what this module does",
    "long_description": "Detailed description of the module's features and purpose.",
}

File: blueprint-go/templates/new-module/manifest.go
package modulename

// Manifest contains module metadata for the loader
var Manifest = ModuleManifest{
	Name:            "ModuleName",
	Version:         "1.0.0",
	MainRoute:       "/modulename",
	Type:            "App", // or "System"
	Depends:         []string{"core"},
	IconClass:       "fa-solid fa-cube",
	Color:           "#007bff",
	Description:     "Short description of your module",
	LongDescription: "Detailed description of what this module does and its features.",
}

// ModuleManifest defines the structure for module metadata
type ModuleManifest struct {
	Name            string   `json:"name"`
	Version         string   `json:"version"`
	MainRoute       string   `json:"main_route"`
	Type            string   `json:"type"` // "App" or "System"
	Depends         []string `json:"depends"`
	IconClass       string   `json:"icon_class"`
	Color           string   `json:"color"`
	Description     string   `json:"description"`
	LongDescription string   `json:"long_description"`
}
